src/firewall/core/rich.py

```diff
@@ -328,6 +328,8 @@ def _import_from_string(self, rule_str):
         if not rule_str:
             raise FirewallError(errors.INVALID_RULE, 'empty rule')
 
+        rule_str = functions.stripCharacters(rule_str)
```

Comment:
Should call `stripNonPrintableCharacters()` here as that's actually what the function is actually doing.